public/components/common/form/hooks.test.tsx

```diff
@@ -1,178 +1,215 @@
+          value: changedValue2,
+        },
+      });
+    });
+
+    // assert changed state
+    expect(result.current.fields.text1.changed).toBe(true);
+    expect(result.current.fields.text1.error).toBeUndefined();
+    expect(result.current.fields.text1.type).toBe(fieldType);
+    expect(result.current.fields.text1.value).toBe(changedValue2);
+    expect(result.current.fields.text1.initialValue).toBe(initialFieldValue);
+
+    // done changes
+    act(() => {
+      result.current.doChanges();
+    });
+
+    // assert do changes state
+    expect(result.current.fields.text1.changed).toBe(false);
+    expect(result.current.fields.text1.error).toBeUndefined();
+    expect(result.current.fields.text1.type).toBe(fieldType);
+    expect(result.current.fields.text1.value).toBe(changedValue2);
+    expect(result.current.fields.text1.initialValue).toBe(changedValue2);
+  });
+
+  it(`[hook] useForm lifecycle. Set the initial value. Change the field value to invalid value`, async () => {
+    const initialFieldValue = 'test';
+    const fieldType = 'text';
+
+    const initialFields: FormConfiguration = {
+      text1: {
+        type: fieldType,
+        initialValue: initialFieldValue,
+        validate: (value: string): string | undefined =>
+          value.length ? undefined : `Validation error: string can be empty.`,
+      },
+    };
+
+    const { result } = renderHook(() => useForm(initialFields));
+
+    // assert initial state
+    expect(result.current.fields.text1.changed).toBe(false);
+    expect(result.current.fields.text1.error).toBeUndefined();
+    expect(result.current.fields.text1.type).toBe(fieldType);
+    expect(result.current.fields.text1.value).toBe(initialFieldValue);
+    expect(result.current.fields.text1.initialValue).toBe(initialFieldValue);
+    expect(result.current.fields.text1.onChange).toBeDefined();
+
+    // change the input
+    const changedValue = '';
+    act(() => {
+      result.current.fields.text1.onChange({
+        target: {
+          value: changedValue,
+        },
+      });
+    });
+
+    // assert changed state
+    expect(result.current.fields.text1.changed).toBe(true);
+    expect(result.current.fields.text1.error).toBeTruthy();
+    expect(result.current.fields.text1.type).toBe(fieldType);
+    expect(result.current.fields.text1.value).toBe(changedValue);
+    expect(result.current.fields.text1.initialValue).toBe(initialFieldValue);
+  });
+
+  it('[hook] useForm. Verify the hook behavior when receives a custom field type', async () => {
+    const CustomComponent = (props: any) => {
+      const { onChange, field, initialValue } = props;
+      const [value, setValue] = useState(initialValue || '');
+
+      const handleOnChange = (e: any) => {
+        setValue(e.target.value);
+        onChange(e);
+      };
+
+          return (
+            <>
+              {field}
+              <input type='text' value={value} onChange={handleOnChange} />
+            </>
+          );
+      };
```

Comment:
**nitpick**: this test file is related to the hook and it doesn't use any ReactJS component. You could simplify the test.